packages/cli/docs-preview/src/previewDocs.ts

```diff
@@ -142,3 +168,28 @@ class ReferencedAPICollector {
         return this.apis;
     }
 }
+
+class ReferencedAPICollectorV2 {
+    private readonly apis: Record<APIDefinitionID, FdrAPI.api.latest.ApiDefinition> = {};
+
+    constructor(private readonly context: TaskContext) {}
+
+    public addReferencedAPI({ api }: { api: FdrAPI.api.latest.ApiDefinition }): APIDefinitionID {
+        try {
+            this.apis[api.id] = api;
+            return api.id;
+        } catch (e) {
+            // Print Error
+            const err = e as Error;
+            this.context.logger.error(`Failed to read referenced API: ${err?.message} ${err?.stack}`);
```

Comment:
what does this error message look like? if its going to print out something thats hard for a user to diagnose it might be better to only print this on `--debug`

Reply:
Or something more human readable `Failed to read OpenAPI spec at /some/path/... Please file an issue`